main/midi/midiPlayer.c

```diff
@@ -1198,23 +1255,15 @@ int32_t midiPlayerStep(midiPlayer_t* player)
         return 0;
     }
 
-    bool checkEvents = false;
+    bool checkEvents = true;
     if (player->mode == MIDI_FILE)
     {
         if (!player->eventAvailable)
         {
             player->eventAvailable = midiNextEvent(&player->reader, &player->pendingEvent);
         }
 
-        if (player->eventAvailable)
-        {
-            if (player->pendingEvent.absTime
-                <= SAMPLES_TO_MIDI_TICKS(player->sampleCount, player->tempo, player->reader.division))
-            {
-                checkEvents = true;
-            }
-        }
-        else
+        if (!player->eventAvailable)
         {
             ESP_LOGI("MIDI", "Done playing file!");
             midiSongEnd(player);
```

Comment:
`checkEvents` should be set to `false` here once the song has ended so it doesn't keep trying to check events later.

Reply:
Fixed